packaging/deb/package.targets

```diff
@@ -0,0 +1,198 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Import Project='package.props' />
+
+  <Target Name="InitializeDotnetDebTool">
+    <Exec Command="$(DotnetToolCommand) pack $(dotnetDebToolSource)/project.json --output $(PackagesOutputDir)intermediate/ --version-suffix $(VersionSuffix)"/>
+
+    <ItemGroup>
+      <pkFiles Include="$(PackagesOutputDir)intermediate/dotnet-deb-tool.*.nupkg" 
+               Exclude="$(PackagesOutputDir)intermediate/*.symbols.nupkg" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(pkFiles)" DestinationFolder="$(PackagesOutputDir)" />
+
+    <RemoveDir Condition="Exists('$(consumingProjectDirectory)')"
+               Directories="$(consumingProjectDirectory)" />
+    
+    <MakeDir Directories="$(consumingProjectDirectory)" />
+
+    <MakeDir Condition="!Exists('$(PackagesIntermediateDir)')"
+             Directories="$(PackagesIntermediateDir)" />
+
+    <ItemGroup>
+      <debJson Include='{' />
+      <debJson Include='  "version": "1.0.0-*",' />
+      <debJson Include='  "name": "$(toolConsumerProjectName)",' />
+      <debJson Include='  "frameworks": { "netcoreapp1.0": { } },' />
+      <debJson Include='  "tools": { "$(dotnetDebToolPackageId)": "$(dotnetDebToolVersion)" }' />
+      <debJson Include='}' />
+    </ItemGroup>
+
+    <WriteLinesToFile
+      File="$(consumingProjectDirectory)\project.json"
+      Lines="@(debJson)"
+      Overwrite="true" />
+
+    <Exec Command="$(DotnetToolCommand) restore -f $(dotnetDebToolPackageSource) --disable-parallel"
+          WorkingDirectory="$(consumingProjectDirectory)" />
+  
+  </Target>
+
+  <Target Name="GenerateDebs" 
+          DependsOnTargets="InitializeDotnetDebTool;GenerateSharedHostDeb;GenerateHostFxrDeb;GenerateSharedFrameworkDeb"
+          Condition="'$(BuildDebPkg)'=='true' and '$(TargetArchitecture)' == 'x64'" />
```

Comment:
a flag passed to when invoking run.exe.
There was an issue open to make building deb packages optional